### ISSUE TITLE
Train loss not lower than 12

### ISSUE THREAD

**Issue by tsungruihon** [mozilla/DeepSpeech#2349]
For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).

If you've found a bug, or have a feature request, then please create an issue with the following information:

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: Docker
- **TensorFlow version (use command below)**: b'v1.13.1-0-g6612da8951' 1.13.1
- **Python version**:  python3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: Docker
- **GPU model and memory**: Docker
- **Exact command to reproduce**: 

You can obtain the TensorFlow version with

```bash
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
```

Please describe the problem clearly. Be sure to convey here why it's a bug or a feature request.

Include any logs or source code that would be helpful to diagnose the problem. For larger logs, link to a Gist, not a screenshot. If including tracebacks, please include the full traceback. Try to provide a reproducible test case.
<hr>

May i ask whether is normal or could accept that the training loss not lower than 12 ? 
The data is Chinese pinyin format, like `lv4 shi4 yang2 chun1 yan1 jing3 da4 kuai4 wen2 zhang1 de5 di3 se4 si4 yue4 de5 lin2 luan2 geng4 shi4 lv4 de5 xian1 huo2 xiu4 mei4 shi1 yi4 ang4 ran2`

```
Epoch 10 |   Training | Elapsed Time: 4:14:48 | Steps: 30125 | Loss: 12.994944
Epoch 10 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.219204 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 11 |   Training | Elapsed Time: 4:33:18 | Steps: 30125 | Loss: 12.839252
Epoch 11 | Validation | Elapsed Time: 0:18:26 | Steps: 1856 | Loss: 17.854801 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 12 |   Training | Elapsed Time: 4:30:30 | Steps: 30125 | Loss: 12.885872
Epoch 12 | Validation | Elapsed Time: 0:19:40 | Steps: 1856 | Loss: 18.398127 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 13 |   Training | Elapsed Time: 4:28:30 | Steps: 30125 | Loss: 12.913519
Epoch 13 | Validation | Elapsed Time: 0:16:31 | Steps: 1856 | Loss: 17.746488 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 14 |   Training | Elapsed Time: 4:15:46 | Steps: 30125 | Loss: 13.052069
Epoch 14 | Validation | Elapsed Time: 0:16:49 | Steps: 1856 | Loss: 17.809107 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 15 |   Training | Elapsed Time: 4:14:39 | Steps: 30125 | Loss: 13.011203
Epoch 15 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.019009 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 16 |   Training | Elapsed Time: 4:29:48 | Steps: 30125 | Loss: 12.950395
Epoch 16 | Validation | Elapsed Time: 0:18:30 | Steps: 1856 | Loss: 17.927670 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 17 |   Training | Elapsed Time: 4:37:58 | Steps: 30125 | Loss: 12.883698
Epoch 17 | Validation | Elapsed Time: 0:19:39 | Steps: 1856 | Loss: 18.019366 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 18 |   Training | Elapsed Time: 4:30:21 | Steps: 30125 | Loss: 13.077873
Epoch 18 | Validation | Elapsed Time: 0:16:30 | Steps: 1856 | Loss: 18.595674 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 19 |   Training | Elapsed Time: 4:17:11 | Steps: 30125 | Loss: 13.077543
Epoch 19 | Validation | Elapsed Time: 0:18:19 | Steps: 1856 | Loss: 18.086610 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 20 |   Training | Elapsed Time: 4:17:33 | Steps: 30125 | Loss: 13.060236
Epoch 20 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.139807 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 21 |   Training | Elapsed Time: 4:27:42 | Steps: 30125 | Loss: 12.917591
Epoch 21 | Validation | Elapsed Time: 0:18:36 | Steps: 1856 | Loss: 17.726031 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
Epoch 22 |   Training | Elapsed Time: 4:33:30 | Steps: 30125 | Loss: 12.972870
Epoch 22 | Validation | Elapsed Time: 0:19:48 | Steps: 1856 | Loss: 17.455123 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
```

**Comment by kdavis-mozilla** [comment]
The loss depends on the language, the noise in the data set, the original audio format.... Basically there is no general rule.

Also, as this is not a bug or feature request could you move the discussion to [discourse](https://discourse.mozilla.org/c/deep-speech). Thanks.

**Comment by tsungruihon** [comment]
thanks @kdavis-mozilla .

**Comment by iekhwang** [comment]
> For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).
> 
> If you've found a bug, or have a feature request, then please create an issue with the following information:
> 
> * **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: No
> * **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
> * **TensorFlow installed from (our builds, or upstream TensorFlow)**: Docker
> * **TensorFlow version (use command below)**: b'v1.13.1-0-g6612da8951' 1.13.1
> * **Python version**:  python3.6
> * **Bazel version (if compiling from source)**:
> * **GCC/Compiler version (if compiling from source)**:
> * **CUDA/cuDNN version**: Docker
> * **GPU model and memory**: Docker
> * **Exact command to reproduce**:
> 
> You can obtain the TensorFlow version with
> 
> ```shell
> python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
> ```
> 
> Please describe the problem clearly. Be sure to convey here why it's a bug or a feature request.
> 
> Include any logs or source code that would be helpful to diagnose the problem. For larger logs, link to a Gist, not a screenshot. If including tracebacks, please include the full traceback. Try to provide a reproducible test case.
> 
> May i ask whether is normal or could accept that the training loss not lower than 12 ?
> The data is Chinese pinyin format, like `lv4 shi4 yang2 chun1 yan1 jing3 da4 kuai4 wen2 zhang1 de5 di3 se4 si4 yue4 de5 lin2 luan2 geng4 shi4 lv4 de5 xian1 huo2 xiu4 mei4 shi1 yi4 ang4 ran2`
> 
> ```
> Epoch 10 |   Training | Elapsed Time: 4:14:48 | Steps: 30125 | Loss: 12.994944
> Epoch 10 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.219204 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 11 |   Training | Elapsed Time: 4:33:18 | Steps: 30125 | Loss: 12.839252
> Epoch 11 | Validation | Elapsed Time: 0:18:26 | Steps: 1856 | Loss: 17.854801 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 12 |   Training | Elapsed Time: 4:30:30 | Steps: 30125 | Loss: 12.885872
> Epoch 12 | Validation | Elapsed Time: 0:19:40 | Steps: 1856 | Loss: 18.398127 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 13 |   Training | Elapsed Time: 4:28:30 | Steps: 30125 | Loss: 12.913519
> Epoch 13 | Validation | Elapsed Time: 0:16:31 | Steps: 1856 | Loss: 17.746488 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 14 |   Training | Elapsed Time: 4:15:46 | Steps: 30125 | Loss: 13.052069
> Epoch 14 | Validation | Elapsed Time: 0:16:49 | Steps: 1856 | Loss: 17.809107 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 15 |   Training | Elapsed Time: 4:14:39 | Steps: 30125 | Loss: 13.011203
> Epoch 15 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.019009 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 16 |   Training | Elapsed Time: 4:29:48 | Steps: 30125 | Loss: 12.950395
> Epoch 16 | Validation | Elapsed Time: 0:18:30 | Steps: 1856 | Loss: 17.927670 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 17 |   Training | Elapsed Time: 4:37:58 | Steps: 30125 | Loss: 12.883698
> Epoch 17 | Validation | Elapsed Time: 0:19:39 | Steps: 1856 | Loss: 18.019366 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 18 |   Training | Elapsed Time: 4:30:21 | Steps: 30125 | Loss: 13.077873
> Epoch 18 | Validation | Elapsed Time: 0:16:30 | Steps: 1856 | Loss: 18.595674 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 19 |   Training | Elapsed Time: 4:17:11 | Steps: 30125 | Loss: 13.077543
> Epoch 19 | Validation | Elapsed Time: 0:18:19 | Steps: 1856 | Loss: 18.086610 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 20 |   Training | Elapsed Time: 4:17:33 | Steps: 30125 | Loss: 13.060236
> Epoch 20 | Validation | Elapsed Time: 0:16:46 | Steps: 1856 | Loss: 18.139807 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 21 |   Training | Elapsed Time: 4:27:42 | Steps: 30125 | Loss: 12.917591
> Epoch 21 | Validation | Elapsed Time: 0:18:36 | Steps: 1856 | Loss: 17.726031 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> Epoch 22 |   Training | Elapsed Time: 4:33:30 | Steps: 30125 | Loss: 12.972870
> Epoch 22 | Validation | Elapsed Time: 0:19:48 | Steps: 1856 | Loss: 17.455123 | Dataset: /home/data/total_csv/words_pinyin/data/chn_dev.csv
> ```

Hello @tsungruihon , I have a question about how you generate your THCHS-30 alphabet.txt of pinyin dataset, and what's the format?  Is it [a-z]?

**Comment by tsungruihon** [comment]
@iekhwang yes.
